From c61db8f014fc71da54fd9d1a6346fdcfb7a5382f Mon Sep 17 00:00:00 2001
From: franciscofranco <franciscofranco.1990@gmail.com>
Date: Mon, 16 Dec 2013 14:52:18 +0000
Subject: [PATCH 396/590] Revert "Revert "msm: idle-v7: flush non secure L1
 cache lines for l2 GDHS LPM mode""

This reverts commit 057eaf3ed6fb760707c5428c9c78bfe52b9af9fb.

Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 arch/arm/mach-msm/idle-v7.S |    1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-msm/idle-v7.S b/arch/arm/mach-msm/idle-v7.S
index 6840f1c..07cd8f0 100644
--- a/arch/arm/mach-msm/idle-v7.S
+++ b/arch/arm/mach-msm/idle-v7.S
@@ -110,6 +110,7 @@ ENTRY(msm_pm_collapse)
 	mov	r2, #1
 	and	r1, r2, r1, ASR #30 /* Check if the cache is write back */
 	orr	r1, r0, r1
+	and	r1, r1, #1
 	cmp	r1, #1
 	bne	skip
 	bl	v7_flush_dcache_all
-- 
1.7.9.5

